### ISSUE TITLE
Huffington Post Not accepting all links

### ISSUE THREAD

**Issue by kieranoreilly** [rg3/youtube-dl#4663]
Hi!
I ran this in terminal: 

 ```
youtube-dl http://embed.live.huffingtonpost.com/HPLEmbedPlayer/?segmentId=547cbb3b02a760cf470005b4  --verbose -o tmp3/KYVbVJ.yt</p>
```
The output was: 

 ```
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['http://embed.live.huffingtonpost.com/HPLEmbedPlayer/?segmentId=547cbb3b02a760cf470005b4', '--verbose', '-o', 'tmp3/KYVbVJ.yt']
[debug] Encodings: locale ANSI_X3.4-1968, fs ANSI_X3.4-1968, out ANSI_X3.4-1968, pref ANSI_X3.4-1968
[debug] youtube-dl version 2015.01.07.2
[debug] Python version 2.7.8 - Linux-3.16-0.bpo.2-amd64-x86_64-with-debian-7.7
[debug] exe versions: avconv 11-6, avprobe 11-6, ffmpeg 2.4.2-1, ffprobe 2.4.2-1, rtmpdump 2.4
[debug] Proxy map: {}
[HuffPost] 547cbb3b02a760cf470005b4: Downloading JSON metadata
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/bin/youtube-dl/__main__.py", line 19, in <module>
  File "/usr/bin/youtube-dl/youtube_dl/__init__.py", line 359, in main
  File "/usr/bin/youtube-dl/youtube_dl/__init__.py", line 349, in _real_main
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 1161, in download
  File "/usr/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 588, in extract_info
  File "/usr/bin/youtube-dl/youtube_dl/extractor/common.py", line 246, in extract
  File "/usr/bin/youtube-dl/youtube_dl/extractor/huffpost.py", line 43, in _real_extract
KeyError: u'schedule'
```

My thoughts is the id has changed to a 24 character id so from (?P<id>[0-9a-f]+) to [0-9a-f]{24}. I haven't confirmed this.

**Comment by dstftw** [comment]
Will be fixed in the next version. Thanks for the report.

**Comment by phihag** [comment]
I believe this issue to be fixed in youtube-dl 2015.01.08. See [our FAQ](https://yt-dl.org/update) if you need help updating.
